fmc/src/flow/rt_alias.rs

```diff
@@ -11,40 +11,168 @@ Abstract:
+        // Create initial output
+        let input = DiceInput {
+            cdi: hand_off.fmc_cdi(),
+            subj_priv_key: hand_off.fmc_priv_key(),
+            auth_key_pair: Ecc384KeyPair {
+                priv_key: KeyId::KeyId5,
+                pub_key: Ecc384PubKey::default(),
+            },
+            auth_sn: [0u8; 64],
+            auth_key_id: [0u8; 20],
+            uds_key: hand_off.fmc_cdi(),
+        };
+
+        Ok(input)
     }
 
     /// Extend current and journey PCRs
     ///
     /// # Arguments
     ///
     /// * `env` - FMC Environment
+    /// * `hand_off` - HandOff
     pub fn extend_pcrs(env: &mut FmcEnv, hand_off: &HandOff) -> CaliptraResult<()> {
         extend_current_pcr(env, hand_off)?;
         extend_journey_pcr(env, hand_off)?;
         Ok(())
     }
+
+    /// Permute Composite Device Identity (CDI) using Rt TCI and Image Manifest Digest
+    ///
+    /// # Arguments
+    ///
+    /// * `env` - ROM Environment
+    /// * `hand_off` - HandOff
+    /// * `cdi` - Key Slot to store the generated CDI
+    ///
+    /// # Returns
+    ///
+    /// * `KeyId` - KeySlot containing the DICE CDI
+    fn derive_cdi(env: &mut FmcEnv, hand_off: &HandOff, cdi: KeyId) -> CaliptraResult<KeyId> {
+        // Get the HMAC Key from CDI
+        let key = Hmac384Key::Key(KeyReadArgs::new(cdi));
+
+        // Compose FMC TCI (1. RT TCI, 2. Image Manifest Digest)
+        let mut tci = [0u8; 2 * SHA384_HASH_SIZE];
+        let rt_tci = Tci::rt_tci(env, hand_off);
+        let rt_tci: [u8; 48] = okref(&rt_tci)?.into();
+        tci[0..SHA384_HASH_SIZE].copy_from_slice(&rt_tci);
+
+        let image_manifest_digest: Result<_, CaliptraError> =
+            Tci::image_manifest_digest(env, hand_off);
+        let image_manifest_digest: [u8; 48] = okref(&image_manifest_digest)?.into();
+        tci[SHA384_HASH_SIZE..2 * SHA384_HASH_SIZE].copy_from_slice(&image_manifest_digest);
+
+        // Permute CDI from FMC TCI
+        let data = Hmac384Data::Slice(&tci);
+        let cdi = Crypto::hmac384_mac(env, key, data, cdi)?;
+        Ok(cdi)
+    }
+
+    /// Derive Dice Layer Key Pair
+    ///
+    /// # Arguments
+    ///
+    /// * `env`      - Fmc Environment
+    /// * `cdi`      - Composite Device Identity
+    /// * `priv_key` - Key slot to store the private key into
+    ///
+    /// # Returns
+    ///
+    /// * `Ecc384KeyPair` - Derive DICE Layer Key Pair
+    fn derive_key_pair(
+        env: &mut FmcEnv,
+        cdi: KeyId,
+        priv_key: KeyId,
+    ) -> CaliptraResult<Ecc384KeyPair> {
+        Crypto::ecc384_key_gen(env, cdi, priv_key)
+    }
+
+    /// Generate Local Device ID Certificate Signature
+    ///
+    /// # Arguments
+    ///
+    /// * `env`    - FMC Environment
+    /// * `input`  - DICE Input
+    /// * `output` - DICE Output
+    fn generate_cert_sig(
+        _env: &FmcEnv,
+        _input: &DiceInput,
+        _output: &DiceOutput,
+    ) -> CaliptraResult<()> {
+        Ok(())
```

Comment:
Will this be done in a follow-up PR? If so, might be good to add a TODO comment (And track in a github issue)

Reply:
Yes, actually both Certificate Generation and Updating of FHT will be done in a different PR. You`re totally right, I updated in my latest commit